lib/chef/audit/logger.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright:: Copyright (c) 2014 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'stringio'
+
+class Chef
+  class Audit
+    class Logger
+      def self.puts(message="")
+        if @buffer.nil?
+          @buffer = StringIO.new
+        end
```

Comment:
Ruby lets you do this: `@buffer ||= StringIO.new`, which is equivalent to this if block.